src/test/java/com/aws/greengrass/clientdevices/auth/iot/ThingTest.java

```diff
@@ -50,65 +62,71 @@ void GIVEN_thingNameWithInvalidCharacters_WHEN_Thing_THEN_exceptionThrown() {
 
     @Test
     void GIVEN_invalidVersion_WHEN_Thing_THEN_exceptionThrown() {
-        Assertions.assertThrows(IllegalArgumentException.class, () -> Thing.of(-1, "Thing"));
-        Assertions.assertDoesNotThrow(() -> Thing.of(0, "Thing"));
+        Assertions.assertThrows(IllegalArgumentException.class, () -> Thing.of(-1, mockThingName));
+        Assertions.assertDoesNotThrow(() -> Thing.of(0, mockThingName));
     }
 
     @Test
     void GIVEN_thing_WHEN_attachCertificate_THEN_certAttachedAndDirtyBitSet() {
-        Thing thing = Thing.of(0, "Thing");
-        thing.attachCertificate("cert-id");
+        Thing thing = Thing.of(0, mockThingName);
+        thing.attachCertificate(mockCertId);
 
         assertThat(thing.isModified(), is(true));
-        List<String> certIds = thing.getAttachedCertificateIds();
-        assertThat(certIds, equalTo(Collections.singletonList("cert-id")));
+        Map<String, Instant> certIds = thing.getAttachedCertificateIds();
```

Comment:
This should not be exposed, add methods inside the thing class that can ask questions about the certificates it has.